src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
+    }
 
+    private void ensureCapacity() {
+        if (size == elementData.length) {
+            int newCapacity = elementData.length * 2;
```

Comment:
When increasing the capacity of the stack, you might want to add a check to ensure that the stack does not grow beyond `Integer.MAX_VALUE`. If `elementData.length` is close to `Integer.MAX_VALUE`, multiplying by 2 could cause an overflow.